bin/Cargo.toml

```diff
@@ -76,6 +76,7 @@ ipnet = { workspace = true, features = ["serde"] }
 # rusqlite is actually only needed for test situations, but we need an optional dependency
 # here so we can disable it for MSRV tests (rusqlite only supports latest stable)
 rusqlite = { workspace = true, features = ["bundled", "time"], optional = true }
+signal-hook.workspace = true
```

Comment:
In another project we solved this using `tokio::unix::Signal`. Maybe consider using that instead?